src/services/codefixes/convertPromisesToAwaitAndAsync.ts

```diff
@@ -0,0 +1,161 @@
+namespace ts.codefix {
+    const fixId = "convertPromisesToAwaitAndAsync";
+    const errorCodes = [Diagnostics.This_may_be_converted_to_use_async_and_await.code];
+    registerCodeFix({
+        errorCodes,
+        getCodeActions(context: CodeFixContext) {
+            const changes = textChanges.ChangeTracker.with(context, (t) => convertToAsyncAwait(t, context.sourceFile, context.span.start, context.program.getTypeChecker()));
+            return [createCodeFixAction(fixId, changes, Diagnostics.Convert_to_use_async_and_await, fixId, Diagnostics.Convert_all_to_use_async_and_await)];
+        },
+        fixIds: [fixId],
+        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, err) => convertToAsyncAwait(changes, err.file!, err.start, context.program.getTypeChecker())),

```

Comment:
this does not handle cases where you have multiple return statements to convert in the same function. the `async` change will be added multiple times. you need to remember which functions you have processed and only add the `async` modifier if you have not seen it before.